game-state-service

```
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+from gi.repository import Gio
+from gi.repository import GLib
+from gi.repository import GObject
+
+import dbus
+import dbus.service
+import dbus.mainloop.glib
+import json
+
+DBUS_IFACE = 'com.endlessm.GameStateService'
+DBUS_PATH = '/com/endlessm/GameStateService'
+
+
+class NotFoundException(dbus.DBusException):
+    _dbus_error_name = 'com.endlessm.GameStateService.NotFoundException'
+
+
+class GameState(GObject.Object):
+
+    _MONITOR_RATE = 1000
+
+    def __init__(self):
+        GObject.Object.__init__(self)
+        self._state = {}
+        self._monitor = None
+        self._monitor_changed_id = None
+        self._monitor_restart_id = None
+        self._monitor_start()
+        self._state_from_file()
+
+    def _get_state_file(self):
+        path = GLib.build_filenamev([GLib.get_user_data_dir(), 'state.json'])
+        return Gio.File.new_for_path(path)
+
+    def _monitor_start(self):
+        state_file = self._get_state_file()
+        self._monitor = state_file.monitor_file(
+            Gio.FileMonitorFlags.NONE,
+            None)
+        self._monitor.set_rate_limit(self._MONITOR_RATE)
+        self._monitor_changed_id = self._monitor.connect(
+            'changed',
+            self._monitor_change_cb)
+
+    def _monitor_change_cb(self, monitor, child, other_file, event):
+        if event == Gio.FileMonitorEvent.CHANGES_DONE_HINT:
+            self._state_from_file()
+
+    def _monitor_restart_cb(self):
+        self._monitor_restart_id = None
+        self._monitor.handler_unblock(self._monitor_changed_id)
+        return False
+
+    def _state_from_file(self):
+        state_file = self._get_state_file()
+        try:
+            success, state_data, tag = state_file.load_contents(None)
+            self._state = json.loads(state_data.decode())
+        except:
```

I'm not a big fan of `except: pass`, can you narrow this down to `IOError` or whatever?